parser-typechecker/src/Unison/Codebase/SqliteCodebase/Operations.hs

```diff
@@ -599,13 +600,50 @@ namesAtPath path = do
         Nothing -> error $ "Expected name to be in namespace" <> show (n, reverse reversedPathSegments)
         Just stripped -> (Name.makeRelative stripped, ref)
 
+-- | Update the root namespace names index which is used by the share server for serving api
+-- requests.
+updateNameLookupIndex ::
+  (C.Reference.Reference -> Sqlite.Transaction CT.ConstructorType) ->
+  Path ->
+  -- | "from" branch, if 'Nothing' use the empty branch
+  Maybe BranchHash ->
+  -- | "to" branch, if 'Nothing' use the root.
+  Maybe BranchHash ->
+  Sqlite.Transaction ()
+updateNameLookupIndex getDeclType pathPrefix mayFromBranchHash mayToBranchHash = do
+  fromBranch <- case mayFromBranchHash of
+    Nothing -> pure V2Branch.empty
+    Just fromBH -> Ops.expectBranchByBranchHash fromBH
+  toBranch <- case mayToBranchHash of
+    Nothing -> Ops.expectRootBranchHash >>= Ops.expectBranchByBranchHash
+    Just toBH -> Ops.expectBranchByBranchHash toBH
+  treeDiff <- BranchDiff.diffBranches fromBranch toBranch
+  let namePrefix = case pathPrefix of
+        Path.Empty -> Nothing
+        (p Path.:< ps) -> Just $ Name.fromSegments (p :| Path.toList ps)
+  let BranchDiff.NameChanges {termNameAdds, termNameRemovals, typeNameAdds, typeNameRemovals} = BranchDiff.nameChanges namePrefix treeDiff
+  termNameAddsWithCT <- do
+    for termNameAdds \(name, ref) -> do
+      refWithCT <- addReferentCT ref
+      pure $ toNamedRef (name, refWithCT)
+  Ops.updateNameIndex (termNameAddsWithCT, toNamedRef <$> termNameRemovals) (toNamedRef <$> typeNameAdds, toNamedRef <$> typeNameRemovals)
+  where
+    toNamedRef :: (Name, ref) -> S.NamedRef ref
+    toNamedRef (name, ref) = S.NamedRef {reversedSegments = coerce $ Name.reverseSegments name, ref = ref}
+    addReferentCT :: C.Referent.Referent -> Transaction (C.Referent.Referent, Maybe C.Referent.ConstructorType)
+    addReferentCT referent = case referent of
+      C.Referent.Ref {} -> pure (referent, Nothing)
+      C.Referent.Con ref _conId -> do
+        ct <- getDeclType ref
+        pure (referent, Just $ Cv.constructorType1to2 ct)
+
 -- | Update the root namespace names index which is used by the share server for serving api
 -- requests.
 --
 -- This version should be used if you've already got the root Branch pre-loaded, otherwise
 -- it's faster to use 'updateNameLookupIndexFromV2Branch'
-updateNameLookupIndexFromV1Branch :: Branch Transaction -> Sqlite.Transaction ()
-updateNameLookupIndexFromV1Branch root = do
+initializeNameLookupIndexFromV1Branch :: Branch Transaction -> Sqlite.Transaction ()
```

Comment:
to me, "initialize" kinda sounds like it would either wipe whatever's there first, or fail if the thing exists already, but I think this just adds names to the index, regardless of whatever's there already. is that an issue?

Reply:
Hrmm, yeah I should probably drop or delete the table in here, good catch.